peekingduck/pipeline/nodes/output/screen.py

```diff
@@ -49,19 +53,66 @@ def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
         cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
         cv2.moveWindow(self.window_name, self.window_loc["x"], self.window_loc["y"])
+        self.previous_filename = ""
+        self.previous_width = 0
+        self.previous_height = 0
 
     def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         """Show the outputs on your display"""
-
         img = inputs["img"]
-        if self.window_size["do_resizing"]:
-            img = cv2.resize(
-                img, (self.window_size["width"], self.window_size["height"])
-            )
+        self._set_window_size(inputs["filename"], img)
         cv2.imshow(self.window_name, img)
-
         if cv2.waitKey(1) & 0xFF == ord("q"):
             cv2.destroyWindow(self.window_name)
             return {"pipeline_end": True}
 
         return {"pipeline_end": False}
+
+    def _set_window_size(self, current_filename: str, img: np.ndarray) -> None:
+        """If `do_resizing` option is False, window size will be initialized to the image or
+        video's frame default size for every new video or image.
+
+        If `do_resizing` option is True, window size will be initialized to the config setting's
+        width and height for every new video or image.
+
+        The length of either sides of the display window will be clamped to a lower bound of
+        `MIN_DISPLAY_SIZE`
+
+        Args:
+            current_filename (str): The filename from the `inputs` dictionary
+            img (np.ndarray): The current image. The image will not be changed in this function.
+        """
+        if current_filename != self.previous_filename:
+            # Initialize the window size for every new video
+            if self.window_size["do_resizing"]:
+                # Clamp the sides fo the window_size to have a minimum of MIN_DISPLAY_SIZE
+                self.window_size["width"] = max(
+                    self.window_size["width"], MIN_DISPLAY_SIZE
+                )
+                self.window_size["height"] = max(
+                    self.window_size["height"], MIN_DISPLAY_SIZE
+                )
+                cv2.resizeWindow(
+                    self.window_name,
+                    self.window_size["width"],
+                    self.window_size["height"],
+                )
+                self.previous_width = self.window_size["width"]
+                self.previous_height = self.window_size["height"]
+            else:
+                img_height, img_width, _ = img.shape
+                cv2.resizeWindow(self.window_name, img_width, img_height)
+                self.previous_width = img_width
+                self.previous_height = img_height
+            self.previous_filename = current_filename
```

Comment:
Is there a need to update the width and height values in `self.window_size`? If not, consider something like the following to avoid repeated code:
```python
if self.window_size["do_resizing"]:
    # Clamp the sides fo the window_size to have a minimum of MIN_DISPLAY_SIZE
    width = max(self.window_size["width"], MIN_DISPLAY_SIZE)
    height = max(self.window_size["height"], MIN_DISPLAY_SIZE)
else:
    height, width, _ = img.shape
self.previous_width = width
self.previous_height = height
cv2.resizeWindow(self.window_name, width, height)
self.previous_filename = current_filename
```

Reply:
Amended